concordium-node/src/consensus_ffi/ffi.rs

```diff
@@ -2439,14 +2440,14 @@ impl ConsensusContainer {
         let sender = Box::new(sender);
         let consensus = self.consensus.load(Ordering::SeqCst);
         let mut buf = [0u8; 32];
-        let (block_id_type, block_hash) =
-            crate::grpc2::types::block_hash_input_to_ffi(block_hash).require()?;
+        let bhi = crate::grpc2::types::block_hash_input_to_ffi(block_hash).require()?;
+        let (block_id_type, block_hash) = bhi.to_ptr();
```

Comment:
Consider making this into a helper function as it is repeated a lot.

Reply:
I don't think it will become easier to read if I factor this out, since the function calling this needs ownership of `bhi` to ensure it is kept alive while consensus processes the query.

Reply:
This is repeated a lot, but it is also necessary due to lifetimes. The original `bhi` must be kept alive long enough that is why we cannot simplify this.